src/test_l3_dsVideoDevice.c

```diff
@@ -0,0 +1,328 @@
+    {NULL, -1}
+};
+
+
+// Mapping table for boolean values
+const static mapping_key_t  boolMappingTable[] = {
+    {"true", (int32_t)true},
+    {"false", (int32_t)false},
+    {NULL, -1}
+};
+
+static char* mapToString(const mapping_key_t* mappingTable, int32_t value) {
+    if (mappingTable == NULL)
+    {
+        return NULL;
+    }
+    for (int32_t i = 0; mappingTable[i].String!=NULL; i++) {
+        if (mappingTable[i].value == value) {
+            return mappingTable[i].String;
+        }
+    }
+    return "Unknown value";
+}
+
+
+/*callback*/
+void dsVideoDevice_FrameratePreChange(unsigned int tSecond)
+{
+	UT_LOG_INFO("FrameratePreChange callback tSecond:[%d]",tSecond);
+}
+
+void dsVideoDevice_FrameratePostChange(unsigned int tSecond)
+{
+	UT_LOG_INFO("FrameratePostChange callback tSecond:[%d]",tSecond);
+}
+
+static void dsVideoDevice_getHandle()
+{
+    dsError_t status   = dsERR_NONE;
+    int32_t choice,device;
+
+    for (device = 0; device < gDSvideoDevice_NumVideoDevices; device++) {
+        UT_LOG_INFO("Calling dsGetVideoDevice(index:[%d])",device);
+        status = dsGetVideoDevice(device,&gHandle);
+        UT_LOG_INFO("Result dsGetVideodevice(Handle:[0x%0X]]) dsError_t=[%s]", gHandle, mapToString(dsErrorMappingTable, status));
+        UT_ASSERT_EQUAL(status, dsERR_NONE);
+        if (status != dsERR_NONE) {
+            UT_LOG_ERROR("dsGetVideodevice failed with error: %d", status);
+        }
+    }
+
+}
+
+void dsVideoDevice_Init()
+{
+    dsError_t status   = dsERR_NONE;
+
+    UT_LOG_INFO("Calling dsVideoDeviceInit()");
+    status = dsVideoDeviceInit();
+    UT_LOG_INFO("Result dsVideoDeviceInit() dsError_t=[%s]", mapToString(dsErrorMappingTable, status));
+    UT_ASSERT_EQUAL_FATAL(status, dsERR_NONE);
+    dsVideoDevice_getHandle();
+	dsRegisterFrameratePreChangeCB(dsVideoDevice_FrameratePreChange);
+	dsRegisterFrameratePostChangeCB(dsVideoDevice_FrameratePostChange);
+
+}
+
+void dsVideoDevice_Term()
+{
+    dsError_t status   = dsERR_NONE;
+
+    UT_LOG_INFO("Calling dsVideoDevice_Term()");
+    status = dsVideoDeviceTerm();
+    UT_LOG_INFO("Result dsVideoDeviceTerm() dsError_t=[%s]",  mapToString(dsErrorMappingTable, status));
+    UT_ASSERT_EQUAL_FATAL(status, dsERR_NONE);
+}
+
+void dsVideoDevice_SetZoomMode()
+{
+    dsError_t status   = dsERR_NONE;
+	int32_t choice,i,j;
+
+	UT_LOG_INFO(" \t  Supported Zoom Modes are:");
+	for (i = 0; i < gDSvideoDevice_NumVideoDevices; ++i) {
+	    for (j = 0; j <gDSVideoDeviceConfiguration[i].NoOfSupportedDFCs; ++j) {
+			UT_LOG_INFO("\t%d.  %-20s ", gDSVideoDeviceConfiguration[i].SupportedDFCs[j], mapToString(dsVideoZoomMappingTable, gDSVideoDeviceConfiguration[i].SupportedDFCs[j]));
+        }
+	}
+
+	UT_LOG_INFO("------------------------------------------");
+    UT_LOG_INFO(" Select the Zoom mode:");
+    scanf("%d", &choice);
+
+	UT_LOG_INFO("Calling dsSetDFC(Handle:[0x%0X]],dsVideoZoom_t:[%s])",gHandle, \
```

Comment:
As per L3 C-Test code practices, mention IN and OUT in the log statement in required places. .

Reply:
updated the pull request